### ISSUE TITLE
Support for groups during discovery

### ISSUE THREAD

**Issue by alvacouch** [hydroshare/hydroshare#2603]
The current discovery interface has no capability of searching for a resource that is accessible to a specific group. This is necessary in order to support "group home page" functions.

**Comment by horsburgh** [comment]
Do we need this? When a person goes to a group's page in HydroShare and clicks on the "Resources" tab, they can already see all of the resources that have been shared with a group.

**Comment by alvacouch** [comment]
@horsburgh The question is: should public resources of a group be visible in a similar fashion? 
Suppose you're outside a group and looking for something published by the group. You can't see the group landing page.

**Comment by aphelionz** [comment]
Yes, we need this. If we have organizations publishing resources to HydroShare we will need to segment them into groups and allow faceting for such.

I wouldnt say we need it *now* but this is something that will be helpful down the line.

**Comment by horsburgh** [comment]
Given what you've described, we need to look at this closer. Currently, users can share a resource with a HydroShare group. But, HydroShare groups cannot be authors of or contributors to resources (given current functionality). You can create an author that is an organization, but you enter the information about the organization (you can't pick a HydroShare group). 

So, right now, the only way to support discovery of resources "accessible to a specific group" would be to delve into the access control rules because there isn't anything in the resource metadata that ties the resource to the group. And, private groups are private, so I shouldn't necessarily be able to discover what a private group has access to.

Mark's use case - organizations publishing resources - needs clarification. HydroShare is the publisher. So, is the group an author? If so, can't you just use the existing functionality to specify an organization as an author? Or, is the group just using HydroShare as a place to publish their data (like iUTAH or potentially CZO) - and they want the capability for people to see all of the resources published as part of the iUTAH project or the by the CZO collaborators, etc. In this case do we just need a special use case of group where the resources shared with that group are actually visible to anyone and not just members of the group?

**Comment by eclark-cuahsi** [comment]
Moved to backlog after discussion 1.25.22
